### DISCUSSION TITLE
Cherrypicking roles from EPUB 3 Structural Semantics Vocabulary 1.1

### DISCUSSION THREAD

**Discussion by HadrienGardeur** [readium/guided-navigation#2] [General]
[The initial list of roles for this project](https://github.com/readium/guided-navigation/blob/main/schema/roles.schema.json) is directly based on [DPUB ARIA 1.0](https://www.w3.org/TR/dpub-aria-1.0/), but there are a number of roles present in [EPUB 3 Structual Semantics Vocabulary 1.1](https://www.w3.org/TR/epub-ssv-11) that are not covered by DPUB ARIA.

There's a mix of:

- deprecated roles
- specialized roles
- and roles with something equivalent in ARIA

We need to define our approach for cherrypicking these roles. My current recommandation would be to:

- ignore deprecated roles
- cherrypick specific specialized roles, when they're deemed useful
- and use ARIA instead when there's an equivalent

Since this is mostly useful for SMIL in our case, ideally we would look extract usage data for roles in SMIL but I'm worried that we might not be able to do that. We could also ask specialized organizations producing Media Overlays about their approach.

**Comment by HadrienGardeur** [comment]
Here's a first list of EPUB roles that are not present in [DPUB ARIA 1.0](https://www.w3.org/TR/dpub-aria-1.0/), I've skipped roles for comics since we already have a dedicated discussion for that:

**Document partitions**

- `backmatter`
- `bodymatter`
- `frontmatter`

**Document divisions**

- `division`
- `volume`

**Document sections and components**

- `preamble`

**Document navigation**

- `landmarks`
- `loa`
- `loi`
- `lot`
- `lov`
- `toc-brief` *(draft)*

**Document reference sections**

- `keywords` *(draft)*

**Dictionaries**

- `antonym-group`
- `condensed-entry`
- `def`
- `dicentry`
- `dictionary`
- `etymology`
- `example`
- `gram-info`
- `idiom`
- `part-of-speech`
- `part-of-speech-list`
- `part-of-speech-group`
- `phonetic-transcription`
- `phrase-list`
- `phrase-group`
- `sense-list`
- `sense-group`
- `synonym-group`
- `tran`
- `tran-info`

**Glossaries**

- `glossdef`
- `glossterm`

**Indexes**

- `index-editor-note`
- `index-entry`
- `index-entry-list`
- `index-group`
- `index-headnotes`
- `index-legend`
- `index-locator`
- `index-locator-list`
- `index-locator-range`
- `index-term`
- `index-term-categories`
- `index-term-category`
- `index-xref-preferred`
- `index-xref-related`

**Preliminary sections and components**

- `contributors`
- `copyright-page`
- `halftitlepage`
- `imprimatur`
- `imprint`
- `other-credits`
- `revision-history`
- `seriespage` *(draft)*
- `titlepage`

**Complementary content**

- `case-study` *(draft)*

**Titles and headings**

- `covertitle `
- `fulltitle`
- `halftitle`
- `label` *(draft)*
- `ordinal` *(draft)*
- `title`

**Educational content**

- `learning-objective`
- `learning-objectives` *(draft)*
- `learning-outcome` *(draft)*
- `learning-outcomes` *(draft)*
- `learning-resource`
- `learning-resources` *(draft)*
- `learning-standard` *(draft)*
- `learning-standards` *(draft)*

**Testing**

- `answer` *(draft)*
- `answers` *(draft)*
- `assessment`
- `assessments` *(draft)*
- `feedback` *(draft)*
- `fill-in-the-blank-problem` *(draft)*
- `general-problem` *(draft)*
- `match-problem` *(draft)*
- `multiple-choice-problem` *(draft)*
- `practice` *(draft)*
- `practices` *(draft)*
- `question` *(draft)*
- `true-false-problem` *(draft)*

**Notes and annotations**

- `footnotes `

**Document text**

- `concluding-sentence`
- `keyword`
- `topic-sentence`

**Tables**

- `table`
- `table-row`
- `table-cell`

**Lists**

- `list`
- `list-item`

**Figures**

- `figure`

**Asides**

- `aside`

**Reply by HadrienGardeur** [reply]
Some early thoughts:

- too many roles overall, many of which are overly specific and probably not necessary if you have a term that works as a superset
- most of these roles were added with the idea of structuring a publication/document, rather than for the type of use case that we're interested in supporting with Guided Navigation
- a number of these terms are clearly identified with a Media Overlay usage context, they're IMO the ones that we should focus on for cherrypicking (I'll post a list below)
- and many of these terms seem tied to satellite specs for which we have very little information whether they are implemented and used today

Here's a list of terms that mention Media Overlays specifically:

**Tables**

- `table`
- `table-row`
- `table-cell`

**Lists**

- `list`
- `list-item`

**Figures**

- `figure`

**Asides**

- `aside`

**Reply by HadrienGardeur** [reply]
In addition to the roles that reference Media Overlay, I'm also adding to our current draft all the roles that usually show up in a Navigation Document, as they're more likely to be used than most of these roles:

- `landmarks`
- `loa`
- `loi`
- `lot`
- `lov`

**Reply by llemeurfr** [reply]
I don't find the "chapter" present in most samples in the specification.

**Reply by llemeurfr** [reply]
Before we choose such structural term, and this is a question to Media Overlays implementors, I'm wondering if these indications are really useful in Reading Systems for managing TTS, and how they are used, since this structural information is part of the HTML side of the publication.

**Reply by HadrienGardeur** [reply]
> I don't find the "chapter" present in most samples in the specification.

That's because it's covered in DPUB ARIA and I already imported all of these roles: https://readium.org/guided-navigation/roles#inherited-from-dpub-aria-10

> Before we choose such structural term, and this is a question to Media Overlays implementors, I'm wondering if these indications are really useful in Reading Systems for managing TTS, and how they are used, since this structural information is part of the HTML side of the publication.

I would be careful with scope. The use cases covered by Guided Navigation Documents are now pretty extensive:

- read aloud (TTS)
- media overlays
- reader mode (for FXL and PDF)
- panel by panel navigation
- accessible comics
- accessible audiobooks
